bokeh/charts/builder.py

```diff
@@ -73,30 +76,26 @@ class Builder(HasProps):
     It provides useful methods to be used by the inherited builder classes,
     in order to automate most of the charts creation tasks and leave the
     core customization to specialized builder classes. In that pattern
-    inherited builders just need to provide:
-
-     - the following methods:
-        * _yield_renderers: yields the glyphs to be rendered into the plot (and
-            eventually create the self._legends attribute to be used to
-            create the proper legends when builder is called to build
-            the glyphs on a Chart object
-        * _process_data(optional): Get the input data and calculates the 'data'
-            attribute to be used to calculate the source data
-        * _set_sources(optional): Push data into the self.source attribute
-            (of type ColumnDataSource) and build the proper ranges
-            (self.x_range and self.y_range).
-
-    - the following attributes:
-        x_range:
-        y_range:
-        _legends:
-
-
-    so Builder can use it all to _yield_renderers on a chart when called with the
-    create method.
+    inherited builders just need to provide the following methods:
+
+    * :meth:`~bokeh.charts.builder.Builder.setup` (optional): provides a pre-renderer
+      creation area to introspect properties, setup attributes, etc.
+    * :meth:`~bokeh.charts.builder.Builder.process_data` (optional): provides a
+      pre-renderer creation area to make manipulations to the source data.
+    * :meth:`~bokeh.charts.builder.Builder.yield_renderers`: yields the glyphs to be
+      rendered into the plot. Here you should call the
+      :meth:`~bokeh.charts.builder.Builder.add_glyph` method so that the builder can
+      setup the legend for you.
+    * :meth:`~bokeh.charts.builder.Builder.set_ranges`: setup the ranges for the
+      glyphs. This is called after glyph creation, so you are able to inspect the
+      comp_glyphs for their minimum and maximum values. See the
+      :meth:`~bokeh.charts.builder.Builder.create` method for more information on
+      when this is called and how the builder provides the ranges to the containing
+      :class:`Chart` using the :meth:`Chart.add_ranges` method.
```

Comment:
small thing, I'd list the mandatory methods first and the optional ones later (it may sound a bit picky, sorry :)